env/environment.go

```diff
@@ -8,36 +8,34 @@ import (
 
 // Environment is a map of environment variables, with the keys normalized
 // for case-insensitive operating systems
-type Environment struct {
-	env map[string]string
-}
+type Environment map[string]string
 
-func New() *Environment {
-	return &Environment{env: map[string]string{}}
+func New() Environment {
```

Comment:
If you're wondering why we're changing all of these pointer receivers (`*Environment`) to value receivers (`Environment`), it's because [all maps are kinda pointers to maps](https://dave.cheney.net/2017/04/30/if-a-map-isnt-a-reference-variable-what-is-it) in go, and having pointers-to-pointers-to-maps would add another layer of indirection and make everything a bit annoying.